dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/logger/SdkInternalLogger.kt

```diff
@@ -253,6 +264,22 @@ internal class SdkInternalLogger(
         }
     }
 
+    private fun Map<String, Any?>?.putIfNotExistOrMapIsNull(key: String, value: Float?): MutableMap<String, Any?> {
+        val mutableMap = this?.toMutableMap() ?: mutableMapOf()
+        return mutableMap.apply { putIfNotExist(key, value ?: return@apply) }
+    }
+
+    // This method same as putIfAbsent but putIfAbsent requires SDK version >= 24
+    @Suppress("UnsafeThirdPartyFunctionCall")
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ I think we could move this extension method in the `dd-sdk-internal` module in a `MapExt.kt` file if you really think we will need it throughout the project otherwise I really don't recommend to have an extension function only for this case.

Reply:
It's not worth it, refactored to simpler one